packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -377,63 +515,81 @@ export const toStyles = ( tree, blockSelectors, hasBlockGapSupport ) => {
+				}
+				ruleset =
+					ruleset +
+					`${ duotoneSelector }{${ duotoneDeclarations.join(
+						';'
+					) };}`;
 			}
-			ruleset =
-				ruleset +
-				`${ duotoneSelector }{${ duotoneDeclarations.join( ';' ) };}`;
-		}
 
-		// Process the remaning block styles (they use either normal block class or __experimentalSelector).
-		const declarations = getStylesDeclarations( styles );
-		if ( declarations?.length ) {
-			ruleset = ruleset + `${ selector }{${ declarations.join( ';' ) };}`;
-		}
+			// Process blockGap and layout styles.
+			if ( ROOT_BLOCK_SELECTOR === selector || hasLayoutSupport ) {
+				ruleset += getLayoutStyles( {
+					tree,
+					style: styles,
+					selector,
+					hasBlockGapSupport,
+					hasBlockStylesSupport,
+				} );
+			}
 
-		// Check for pseudo selector in `styles` and handle separately.
-		const psuedoSelectorStyles = Object.entries( styles ).filter(
-			( [ key ] ) => key.startsWith( ':' )
-		);
+			// Process the remaning block styles (they use either normal block class or __experimentalSelector).
+			const declarations = getStylesDeclarations( styles );
+			if ( declarations?.length ) {
+				ruleset =
+					ruleset + `${ selector }{${ declarations.join( ';' ) };}`;
+			}
 
-		if ( psuedoSelectorStyles?.length ) {
-			psuedoSelectorStyles.forEach( ( [ pseudoKey, pseudoRule ] ) => {
-				const pseudoDeclarations = getStylesDeclarations( pseudoRule );
+			// Check for pseudo selector in `styles` and handle separately.
+			const psuedoSelectorStyles = Object.entries( styles ).filter(
+				( [ key ] ) => key.startsWith( ':' )
+			);
 
-				if ( ! pseudoDeclarations?.length ) {
-					return;
-				}
+			if ( psuedoSelectorStyles?.length ) {
+				psuedoSelectorStyles.forEach( ( [ pseudoKey, pseudoRule ] ) => {
+					const pseudoDeclarations =
+						getStylesDeclarations( pseudoRule );
 
-				// `selector` maybe provided in a form
-				// where block level selectors have sub element
-				// selectors appended to them as a comma seperated
-				// string.
-				// e.g. `h1 a,h2 a,h3 a,h4 a,h5 a,h6 a`;
-				// Split and append pseudo selector to create
-				// the proper rules to target the elements.
-				const _selector = selector
-					.split( ',' )
-					.map( ( sel ) => sel + pseudoKey )
-					.join( ',' );
-
-				const psuedoRule = `${ _selector }{${ pseudoDeclarations.join(
-					';'
-				) };}`;
-
-				ruleset = ruleset + psuedoRule;
-			} );
+					if ( ! pseudoDeclarations?.length ) {
+						return;
+					}
+
+					// `selector` maybe provided in a form
+					// where block level selectors have sub element
+					// selectors appended to them as a comma seperated
+					// string.
+					// e.g. `h1 a,h2 a,h3 a,h4 a,h5 a,h6 a`;
+					// Split and append pseudo selector to create
+					// the proper rules to target the elements.
+					const _selector = selector
+						.split( ',' )
+						.map( ( sel ) => sel + pseudoKey )
+						.join( ',' );
+
+					const psuedoRule = `${ _selector }{${ pseudoDeclarations.join(
```

Comment:
Typo: `pseudoRule`

Reply:
Ah, that typo already existed but it's been caught in this diff due to the spacing change. I'll fix the typo while I'm here 😄